cli/urfave/mod.go

```diff
@@ -0,0 +1,165 @@
+// Package urfave provides a cli builder implementation based on the urfave/cli
+// library.
+package urfave
```

Comment:
why name this after urfave, considering that all the exported types, functions, etc. are NOT urfave-dependent? As a matter of fact, if I understand correctly, from this package's consumer's perspective, this package's dependence on urfave is an implementation detail.
We could swap it out for another dependency and the user wouldn't notice.

could we name the package after its function? would `builder` be suited ?

Reply:
Even if the exposed functions are not urfave dependent, we name it after its underlying implementation. A user is then free to choose its implementation based on the implementation details, which can matter, like with could provide a `fake` cli. 
This is how it's done in Dela. This is why we, for example, have `minogrpc` and `minoch`, which both provides the same exported types but with different implementation details. Could say the same with the ordering module: we have a `pow (proof-of-work)` one and a `cosipbft` one.

Reply:
OK, I can accept this.
Could we at least have 'cli' in the package name, like for mino? It'd make the using code more readable (ex. `urcli.NewBuilder()` as opposed to `urfave.NewBuilder()`) 

For reference, I've tried to express my thinking below
---
Architecture is a way of supporting some desirable characteristics of a software system, in a general sense we're managing risks related to its evolution, stability and performance. As an example, we know that the overlay network implementation may need to change, so we design an abstraction for it (mino). We know that ordering will need to change, so we do the same. We want to be able to run experiments, so we make it easy to have multiple swappable implementations.

In the CLI case, do we really expect to have use for multiple implementations? A `fake` CLI seems something that belongs in an integration test package, for instance. On the other hand, does this mean that if we want to change the CLI implementation, we have to go through each and every user of the `urfave` package, to point to a different package in the dependency injection system ? That seems cumbersome and 'urfave' isn't quite explicit as a package name.

Ultimately, it's the CLI so I don't expect there'll be too many places making use of this, and this design might help change the urfave-based CLI programs one-at-a-time if need be. It makes sense.